### PR TITLE
add ib-sriov-cni

### PR DESCRIPTION
**Pull request by zenghui-shi**

### CODE REVIEW

**Comment by zshi-redhat** [comment]
> /hold
> 
> Config looks good, approving to allow merge. The hold allows you to act on rehearsal failures, if needed (feel free to unhold)

Thanks Petr, the reherasal failures were due to old golang version 1.10 is used in Dockerfile, which failed with `-modcache` flag not supported. Fix was proposed and will re-test once fix merged.

btw, in ART build, it will automatically replace the old golang version 1.10 to 1.13, which doesn't have above issue. In origin build, it doesn't do automatic replacement.

**Comment by zshi-redhat** [comment]
@petr-muller I fixed the Dockerfile issue in openshift/ib-sriov-cni/Dockerfile.rhel7, but it seems the CI check in this PR is still using the old content, do you know how I make the CI check to use latest change?

The change I did in openshift/ib-sriov-cni/Dockerfile.rhel7 was:
change: `registry.svc.ci.openshift.org/ocp/builder:golang-1.10`
to: `registry.svc.ci.openshift.org/ocp/builder:golang-1.13`

but in one of the CI failures, it still uses `golang-1.10`: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_release/9524/rehearse-9524-pull-ci-openshift-ib-sriov-cni-release-4.6-images/5/build-log.txt

**Comment by petr-muller** [comment]
> but in one of the CI failures, it still uses golang-1.10: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_release/9524/rehearse-9524-pull-ci-openshift-ib-sriov-cni-release-4.6-images/5/build-log.txt

This is a job that runs for your `release-4.6` branch, which does not seem to have your Dockerfile change. 

The simplest fix here is only add CI configs (and jobs) for `master` branch, not the release ones. The release ones will be created automatically after you merge the `master` one, and our systems will start automatically fast-forwarding your code form master to `release-4.6` and `release-4.7` branches, so everything will work.

**Comment by zshi-redhat** [comment]
> > but in one of the CI failures, it still uses golang-1.10: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_release/9524/rehearse-9524-pull-ci-openshift-ib-sriov-cni-release-4.6-images/5/build-log.txt
> 
> This is a job that runs for your `release-4.6` branch, which does not seem to have your Dockerfile change.

Thanks Petr!
Yes, I missed that, it runs on the release branches. :-)

> 
> The simplest fix here is only add CI configs (and jobs) for `master` branch, not the release ones. The release ones will be created automatically after you merge the `master` one, and our systems will start automatically fast-forwarding your code from master to `release-4.6` and `release-4.7` branches, so everything will work.

I'll try to fast-forward the same dockerfile fix to release branches, and re-run the CI check.

**Comment by zshi-redhat** [comment]
> Oh, this needs @smarterclayton for the image mirroring part
> 
> https://github.com/openshift/release/blob/master/core-services/image-mirroring/openshift/OWNERS

@smarterclayton PTAL

**Comment by smarterclayton** [comment]
Your naming is `lb-sriov-cni`?  What does that mean?  Naming needs to be clear on images in the payload.

**Comment by smarterclayton** [comment]
You should be following the naming pattern `sriov-<thing>-cni` - what does `ib` mean?

**Comment by zshi-redhat** [comment]
> You should be following the naming pattern `sriov-<thing>-cni` - what does `ib` mean?

`ib` is abbreviation of [InfiniBand](https://en.wikipedia.org/wiki/InfiniBand).
There is a `sriov-cni` which is for Ethernet network device. This one is for InfiniBand SR-IOV device.
I'll make a change to follow the naming pattern `sriov-infiniband-cni`.

**Comment by zshi-redhat** [comment]
rebase and fix confliction

File: core-services/image-mirroring/openshift/mapping_origin_4_7 [hunk 1]
```diff
@@ -199,6 +199,7 @@ registry.svc.ci.openshift.org/ocp/4.7:service-ca-operator quay.io/openshift/orig
 registry.svc.ci.openshift.org/ocp/4.7:service-catalog quay.io/openshift/origin-service-catalog:4.7 quay.io/openshift/origin-service-catalog:4.7.0
 registry.svc.ci.openshift.org/ocp/4.7:special-resource-operator quay.io/openshift/origin-special-resource-operator:4.7 quay.io/openshift/origin-special-resource-operator:4.7.0
 registry.svc.ci.openshift.org/ocp/4.7:sriov-cni quay.io/openshift/origin-sriov-cni:4.7 quay.io/openshift/origin-sriov-cni:4.7.0
+registry.svc.ci.openshift.org/ocp/4.7:sriov-infiniband-cni quay.io/openshift/origin-sriov-infiniband-cni:4.7 quay.io/openshift/origin-sriov-infiniband-cni:4.7.0
```
Comment (smarterclayton) [inline core-services/image-mirroring/openshift/mapping_origin_4_7:202]: this file is sorted, so this line should be after sriov-dp-admission-controller

Comment (zshi-redhat) [reply core-services/image-mirroring/openshift/mapping_origin_4_7]: fixed for both 4_7 and 4_6

**Pull request merged**